Wei li add and delete patient feature

tanweili opened the pull request.

tanweili committed:
Trim string outputs of readCommand() and readParameters()

For proper parsing of commands and parameters and keyword
recognition that may not work due to the extra space
character.

tanweili committed:
Implement basic switch-case statement in Manager class

To have a standard syntax in how we identify the command for now.

Fixes #28

shxr3f commented on src/main/java/seedu/duke/Manager.java:
```diff
@@ -19,6 +17,26 @@ public void runLoop() {
+                break;
+            case "view medicine":
+                break;
+            case "help":
+                ui.printHelp();
```
Needs break after this case.

tanweili replied:
Yep thanks.

shxr3f reviewed (changes requested):
Generally seems fine, except for the missing break.

tanweili committed:
Implement basic validation template for parseAddPatient

Also added break keyword in Manager to prevent fallthrough

tanweili commented:
Should we use regular expression for validation for some of the parameters like what I written above?

tanweili committed:
Implement addPatient Method in Command Class

tanweili committed:
Implement deletePatient method in Command

Add getSize() and removePatient() methods in PatientList to support
deletePatient.
Array of patients in PatientList class is now private.

tanweili committed:
Add add patient and delete patient case statement logic

tanweili committed:
Refactor code for validation in parseAddPatient into various methods

Methods to validate each piece of information can be reused for adding
Doctors and Medicines.

kktai1512 commented on src/main/java/seedu/duke/assets/Patient.java:
```diff
@@ -7,17 +7,17 @@
     String address;
     String dob;
-    String dateAdmission;
+    String AdmissionDate;
```
is  camelCase naming more appropriate?

tanweili replied:
Yeah thanks for pointing out

kktai1512 commented on src/main/java/seedu/duke/helper/Parser.java:
```diff
@@ -19,4 +22,71 @@
+        }
+    }
+
+    private static boolean validateGender(String gender) {
```
am thinking perhaps can abstract out all the validators to a validator class separate from parser

tanweili replied:
Also can. But I think let's focus on getting all the three Patient, Medicine and Doctor up first before thinking about more OOP refactoring

wraineflores commented on src/main/java/seedu/duke/assets/PatientList.java:
```diff
@@ -4,8 +4,7 @@
 
-    ArrayList<Patient> patientList;
-
+    private ArrayList<Patient> patientList;
```
patientLists for arrays. Does it need to be plural, right?

shxr3f replied:
I think naming it patients would be more appropriate

tanweili replied:
Yeah I'll change to patients as well. Thanks

wraineflores commented on src/main/java/seedu/duke/helper/Command.java:
```diff
@@ -14,4 +14,25 @@ public void viewPatient(PatientList patientList, String parameters) {
     }
 
+    public void addPatient(PatientList patientList, String parameters) {
+        String[] parametersArray = Parser.parseAddPatient(parameters);
```
Can we change the name to be plural since it is an array? Like arrayOfInputParameters or something?

wraineflores commented on src/main/java/seedu/duke/helper/Parser.java:
```diff
@@ -19,4 +22,71 @@
     public static String[] commandParser(String userInput) {
```
is it parserCommand instead of commandParser since it is a noun? Or does it need to be changed to sound like a verb such as parseNewInput?

tanweili committed:
Finalize AddPatient method for now

AddPatient is working as long as the input format is correct but
there is a lot of code quality issues to be addressed.

Also the typing in bye will crash the application for now due to
the way we are implementing readParameter now which also needs
to be addressed.

tanweili committed:
Change AdmissionDate to admissionDate in Patient

Code quality as pointed by Wraine.

tanweili committed:
Update input.txt and EXPECTED.txt

tanweili committed:
Update checkstyle for && in Parser

Based on Gradle error message.

tanweili committed:
Update EXPECTED.txt 2

To fit ACTUAL.txt.

tanweili commented:
I think this PR is getting too long. Pls help to review and see if can merge coz I have done some edits to common lines of code that probably will affect y'all parts. I'll try to be more specific in my PR from now.

shxr3f reviewed (approved):
I think it looks fine overall, if there are any changes to code structure we can edit it in future PRs. Good job, I think can merge!
